libnmstate/nm/dns.py

```diff
@@ -118,3 +127,25 @@ def get_dns_config_iface_names(acs_and_ipv4_profiles, acs_and_ipv6_profiles):
         if ip_profile.props.dns or ip_profile.props.dns_search:
             iface_names.append(nm_ac.ActiveConnection(ac).devname)
     return iface_names
+
+
+def find_interfaces_for_name_servers(iface_routes):
```

Comment:
This is now pure route helper, as it finds the interfaces of default gateway routes.
You could move the logic to the route module and call it from here. Keeping the route logic in the route domain is a nice separation.

Anyway, I do not know how it will turn out later, so we can re-evaluate this later on.